### DISCUSSION TITLE
Add mp4 support

### DISCUSSION THREAD

**Discussion by jherazob** [chevereto/chevereto#43] [Ideas]
These days image hosting sites allow you to upload short MP4 files instead of inefficient GIFs (and many in fact automatically convert big GIFs into MP4 clips behind the curtain for space reasons, i know Imgur does but pretty sure many others do it too), would be good if this software also added support for these

**Comment by rodber** [comment]
Any idea on size for these videos? 15MB? 50MB?

**Comment by jherazob** [comment]
I know the requirements for sites like Imgur and Reddit are very high (if i recall Imgur only visibly limits clips with audio by length, up to 1 minute, but without sound i think it has some absurd ceiling like 200Mb or something, and Reddit's are even higher even with audio), and i guess in the end the server admin will have the ultimate choice, so my guess is that it should be configurable by the installer. 

This is image hosting so if you wanna host 4K movies in it it's your problem (and likely *heavily* discouraged on the docs), but most people will likely only host short clips which is what image hosting sites do. Hell, they do it to save bandwidth, take a huge gif and turn it into an mp4 and it's suddenly like 1/10th the size, that's why hosting sites started doing it. Imgur in particular i know that internally, when they get a gif bigger than 2Mb, they automatically convert it to mp4 and only serve that ("gifv", an mp4 enveloped by a minimal html) unless explicitly asked for the gif.

So yeah, likely user-configurable with some sane default.

**Comment by rodber** [comment]
Client applications often compress image/video on client side, that way processing gets cheaper on the server. We don't have that layer yet.

If this is straight h264 the job required by the server is minimum (take a frame and done) but for other formats it gets harder as it will require processing.

In other words, yay but only for h264.

**Comment by jherazob** [comment]
Checked and yeah, h264 seems to be the codec used in non-video sites, confirmed that both Reddit and Imgur use it, and i'd bet the same is true of others like Tumblr and others. This also seems to be well supported in both desktop and mobile. So yeah, your guess seems correct, which is good.

**Comment by rodber** [comment]
Video support has been added in v4.1.0
https://github.com/chevereto/chevereto/releases/tag/4.1.0
